### PR TITLE
add trace to rake seed on test

### PR DESCRIPTION
**Pull request by sureshc**
add trace to rake seed on test and send output to Slack

https://trello.com/c/6V3kp5mW/28-add-trace-to-seed-commands-on-test

### CODE REVIEW

File: lib/rake/build.rake [hunk 1]
```diff
@@ -81,7 +81,7 @@ namespace :build do
         else
           ChatClient.log 'Seeding <b>dashboard</b>...'
           ChatClient.log 'consider setting "skip_seed_all" in locals.yml if this is taking too long' if rack_env?(:development)
-          RakeUtils.rake 'seed:all'
+          RakeUtils.rake 'seed:all', (rack_env?(:test) ? '--trace' : nil)
```
Comment (sureshc) [inline lib/rake/build.rake:84]: should I include `ENV['CI']` as well?  My understanding is that would detect the CircleCI environments

**Comment by sureshc** [comment]
Am I correct in assuming that this line ensures the output of the pegasus and dashboard seed tasks will be logged to Slack?  

https://github.com/code-dot-org/code-dot-org/blob/23bf872d0ce27d60f290c1c6457193f77d50a379/lib/rake/ci.rake#L43

File: lib/rake/build.rake (hunk 1 above)
Comment (balderdash) [reply lib/rake/build.rake:84]: `RACK_ENV` is `test` in our CircleCI builds as well: https://github.com/code-dot-org/code-dot-org/blob/cf86dfa5e97a1ff3c7f217216df38654d84d4c75/.circleci/config.yml#L14-L16

Comment (sureshc) [reply lib/rake/build.rake:84]: thanks!

**Comment by wjordan** [comment]
> Am I correct in assuming that this line ensures the output of the pegasus and dashboard seed tasks will be logged to Slack?

The `ChatClient.wrap` part only ensures that the 'start' and 'end' events of the block will be logged to Slack, I don't think there's any other output-capturing logic in its implementation that would do anything further than that. This means that whether the output of the seed tasks will be logged to slack and/or s3 depends on what happens in its implementation.

Here's a dig through the chain of events (with the help of GitHub's code linking), starting with that `ci::build` Rake task invoked:

https://github.com/code-dot-org/code-dot-org/blob/ee3735face43fa9d5bdc126b8000f53ca0e69fc5/lib/rake/ci.rake#L38-L45

which invokes the `build` Rake task:

https://github.com/code-dot-org/code-dot-org/blob/3c7bca1a59e5a7c561c5db683d0916e401b700c6/lib/rake/build.rake#L154-L157

which invokes `build:all`:

https://github.com/code-dot-org/code-dot-org/blob/3c7bca1a59e5a7c561c5db683d0916e401b700c6/lib/rake/build.rake#L145-L151

which invokes `build:dashboard`, which calls the seed task:

https://github.com/code-dot-org/code-dot-org/blob/3c7bca1a59e5a7c561c5db683d0916e401b700c6/lib/rake/build.rake#L84

Which is done by `RakeUtils#rake`:

https://github.com/code-dot-org/code-dot-org/blob/cf86dfa5e97a1ff3c7f217216df38654d84d4c75/lib/cdo/rake_utils.rb#L278-L280

https://github.com/code-dot-org/code-dot-org/blob/cf86dfa5e97a1ff3c7f217216df38654d84d4c75/lib/cdo/rake_utils.rb#L143-L145

https://github.com/code-dot-org/code-dot-org/blob/cf86dfa5e97a1ff3c7f217216df38654d84d4c75/lib/cdo/rake_utils.rb#L87-L97

The error-case in this `system` method looks like it [`raise`](http://ruby-doc.org/core-2.2.0/Kernel.html#method-i-raise)s an exception with the `output` from `system__` provided as the callback information. For some reason that output isn't making its way either to the S3 log or to Slack.

**Comment by sureshc** [comment]
A little higher up the call chain, the `main` method in `aws/ci_build` invokes this nifty utility which seems to capture all standard out and standard error.  I'm sorting through `ci_build` to see if it collects up that output and sends it to S3/Slack:

https://github.com/code-dot-org/code-dot-org/blob/673e6574461e14ad1b91b46f4b652c154db1b1fd/lib/cdo/rake_utils.rb#L331-L361

**Comment by wjordan** [comment]
Yeah, I added `capture` a while back (#11860) as part of an earlier attempt to unwind a bit of this- I think it ended up being only partly successful, I don't know exactly where parts of the output are getting left out in the failure case.

A bigger project would be to figure out a way to start adding unit-test coverage to parts of these scripts so we can test/verify bugs quicker and with more confidence. That be quite tricky to set up though, which is why we haven't done it yet :sweat_smile:

**Comment by sureshc** [comment]
Possibly the cases where a rake task is being invoked with `system()` is causing them to be run in a different process with standard out not being captured (https://stackoverflow.com/questions/7212573/when-to-use-each-method-of-launching-a-subprocess-in-ruby/7263556#7263556)?  Another reason to invoke rake tasks directly via ruby from other rake tasks and other ruby logic.

**Pull request merged**